src/users/sign-up/sign-up.service.ts

```diff
@@ -0,0 +1,45 @@
+import bcrypt from 'bcrypt'
+import env from '@/configs/env'
+import { JwtSignAdapter } from '@/infra/cryptography/jwt-sign-adapter'
+import { UuidAdapter } from '@/infra/uuid-adapter/uuid-adapter'
+import { PrismaService } from '@/shared/prisma/prisma.service'
+import { Injectable } from '@nestjs/common'
+import { SignUpWithEmailDto } from './sign-up-with-email-dto'
+
+const SALT_ROUNDS = 10
+
+@Injectable()
+export class SignUpService {
+  constructor (private readonly prismaService: PrismaService) {}
+
+  async create ({ email, password }: SignUpWithEmailDto): Promise<{ token: string } | undefined> {
+    console.log('Aqui')
+
+    try {
+      const signUpData = new SignUpWithEmailDto(email, password)
+
+      const user = await this.prismaService.userWithEmail.findUnique({ where: { email } })
+
+      if (user) {
+        throw new Error('User already exists')

```

Comment:
Não consegui identificar porque independente do que acontece nessa requisição, tudo retorna status 201.

Reply:
Então nesse caso, a melhor forma e você fazer um try cath no Controller e lá no controller retornar o Error. O Nestjs já tem algumas classes de error. 

Docs: https://docs.nestjs.com/exception-filters#built-in-http-exceptions